lib/Runtime/Language/JavascriptStackWalker.cpp

```diff
@@ -616,7 +616,34 @@ namespace Js
         }
         return nullptr;
     }
-
+#if ENABLE_NATIVE_CODEGEN
+    void JavascriptStackWalker::WalkAndClearInlineeFrameCallInfoOnException()
+    {
+        // Walk the stack and when we find the first JavascriptFrame, we clear the inlinee's callinfo for this frame
+        // It is sufficient we stop at the first Javascript frame which had the enclosing try-catch
+        while (this->Walk(true))
+        {
+            if (this->IsJavascriptFrame())
+            {
+                if (!this->isNativeLibraryFrame)
+                {
+                    if (HasInlinedFramesOnStack())
+                    {
+                        for (int index = inlinedFrameWalker.GetFrameCount() - 1; index >= 0; index--)
+                        {
+                            auto inlinedFrame = inlinedFrameWalker.GetFrameAtIndex(index);
+                            inlinedFrame->callInfo.Clear();
+                        }
```

Comment:
This works for now, but we'll have to keep in mind to fix this when we start inlining functions with try-catch-finally. If I read this correctly, with functions with EH inlined, this loop might clear callInfo on more inlined frames than intended.

Reply:
We clear the all inlineFrames from the point of exception till the enclosing try catch.